Content.Shared/Ninja/Systems/SharedNinjaGlovesSystem.cs

```diff
@@ -0,0 +1,181 @@
+using Content.Shared.Actions;
+using Content.Shared.CombatMode;
+using Content.Shared.Doors.Components;
+using Content.Shared.DoAfter;
+using Content.Shared.Examine;
+using Content.Shared.Hands.Components;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Interaction;
+using Content.Shared.Interaction.Components;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Inventory.Events;
+using Content.Shared.Ninja.Components;
+using Content.Shared.Popups;
+using Content.Shared.Research.Components;
+using Content.Shared.Timing;
+using Content.Shared.Toggleable;
+using Robust.Shared.Timing;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Content.Shared.Ninja.Systems;
+
+public abstract class SharedNinjaGlovesSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] protected readonly SharedAppearanceSystem Appearance = default!;
+    [Dependency] private readonly SharedCombatModeSystem _combatMode = default!;
+    [Dependency] protected readonly SharedDoAfterSystem _doAfter = default!;
+    [Dependency] protected readonly SharedInteractionSystem Interaction = default!;
+    [Dependency] private readonly SharedSpaceNinjaSystem _ninja = default!;
+    [Dependency] protected readonly SharedPopupSystem Popup = default!;
+    [Dependency] private readonly UseDelaySystem _useDelay = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<NinjaGlovesComponent, GetItemActionsEvent>(OnGetItemActions);
+        SubscribeLocalEvent<NinjaGlovesComponent, ExaminedEvent>(OnExamined);
+        SubscribeLocalEvent<NinjaGlovesComponent, GotUnequippedEvent>(OnUnequipped);
+
+        // TODO: maybe move into r&d server???
+        SubscribeLocalEvent<NinjaDownloadComponent, InteractionAttemptEvent>(OnDownload);
+    }
+
+    /// <summary>
+    /// Disable glove abilities and show the popup if they were enabled previously.
+    /// </summary>
+    public void DisableGloves(EntityUid uid, NinjaGlovesComponent comp)
+    {
+        if (comp.User != null)
+        {
+            var user = comp.User.Value;
+            comp.User = null;
+            Dirty(comp);
+
+            Appearance.SetData(uid, ToggleVisuals.Toggled, false);
+            RemComp<InteractionRelayComponent>(user);
+            Popup.PopupClient(Loc.GetString("ninja-gloves-off"), user, user);
+
+            RemComp<BatteryDrainerComponent>(user);
+            RemComp<EmagProviderComponent>(user);
+            RemComp<StunProviderComponent>(user);
+        }
+    }
+
+    /// <summary>
+    /// Adds the toggle action when equipped.
+    /// Since the event does not pass user this can't be nice and just not add the action if it isn't a ninja wearing but oh well.
+    /// </summary>
+    private void OnGetItemActions(EntityUid uid, NinjaGlovesComponent comp, GetItemActionsEvent args)
+    {
+        args.Actions.Add(comp.ToggleAction);
+    }
```

Comment:
Can't you just add a whitelist to actions?

Reply:
action whitelist is for target of TargetedAction, but i can just use args.User from stealth clothing pr and check there

Reply:
args.User is checked for all actions